common: ocispec: handle balancing policy as enum

mlohvynenko opened the pull request.

mlohvynenko committed:
common: ocispec: handle balancing policy as enum

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mykola-kobets-epam reviewed (approved):
`Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>`

MykolaSuperman reviewed (approved):
`Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>`

al1img reviewed (approved):
`Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>`

mlohvynenko committed:
cmake: exclude tests dirs from coverage

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>

mlohvynenko committed:
cmake: set merge-mode-functions for gcovr

This patch sets gcovr merge-mode-functions to merge-use-line-min.
By default, gcovr errors when it finds multiple functions with
the same name but different line coverage.
This happend for the core library types/common.hpp file,
which is included in multiple translation units.

Signed-off-by: Mykhailo Lohvynenko <mykhailo_lohvynenko@epam.com>
Reviewed-by: Mykola Kobets <mykola_kobets@epam.com>
Reviewed-by: Mykola Solianko <mykola_solianko@epam.com>
Reviewed-by: Oleksandr Grytsov <oleksandr_grytsov@epam.com>
